src/Domain/Movie/MovieRepository.php

```diff
@@ -672,16 +672,19 @@ public function fetchUniqueMoviesPaginated(
 
         return $this->dbConnection->fetchAllAssociative(
             <<<SQL
-            SELECT m.*, mur.rating as userRating
-            FROM movie m
-            JOIN movie_user_watch_dates mh on mh.movie_id = m.id and mh.user_id = ?
-            LEFT JOIN movie_user_rating mur ON mh.movie_id = mur.movie_id and mur.user_id = ?
-            LEFT JOIN movie_genre mg on m.id = mg.movie_id
-            LEFT JOIN genre g on mg.genre_id = g.id
-            $whereQuery
-            GROUP BY m.id, title, release_date, watched_at, rating
-            ORDER BY $sortBySanitized $sortOrder, title asc
-            LIMIT $offset, $limit
+            SELECT * FROM (
+                SELECT m.*, mur.rating as userRating, ROW_NUMBER() OVER(PARTITION BY m.id) rn
+                FROM movie m
+                JOIN movie_user_watch_dates mh on mh.movie_id = m.id and mh.user_id = ?
+                LEFT JOIN movie_user_rating mur on mh.movie_id = mur.movie_id and mh.user_id = ?
+                LEFT JOIN movie_genre mg on m.id = mg.movie_id
+                LEFT JOIN genre g on mg.genre_id = g.id
+                $whereQuery
+                GROUP BY m.id, title, release_date, watched_at, rating
+                ORDER BY $sortBySanitized $sortOrder, title asc
+                LIMIT $offset, $limit
+            ) a
+            WHERE rn = 1
```

Comment:
```suggestion
            ) a
            WHERE rn = 1
            LIMIT $offset, $limit
```

We nearly got it I think :D We need to move the offset and limit out of the subquery, otherwise the subquery returns e.g. the wanted 24 results and than removes the results not matching the where -> we have less results than wanted which causes something like this:
![image](https://github.com/leepeuker/movary/assets/19872901/9eb56f6f-4611-4de7-b07c-f31988a9bb34)

Reply:
hm but this still does not seem right :thinking: 

maybe we just solve this via code instead of query, not as elegant but simpler I guess :D

Reply:
Ah shit via code does not work either because when we get the max amount of results we cannot just remove something from it :hankey:

Reply:
hm, im not getting that bug using sqlite.

the limit line is inside the subquery because that's where all the data is fetched, and i think placing it outside of that would query all of the movies for each page and THEN filter them.

Reply:
which movies are being ignored by the query? i feel like this has to do with the the whole query in general rather than the LIMIT line, unfortunately.

since i'm not running mysql it's probably easier for you to debug which entries are being ignored, also because i don't have access to a computer now.

Reply:
oh ok i see what you mean now, and i was able to replicate the bug with sqlite as well.

you're right about the LIMIT line being the culprit here, it seems that we would have to increment the offset/limit by 1 for each duplicate entry.


i tried applying the fix that you suggested and... it seems to work for me. all 24 slots are taken even with movies that have various watch dates.

Reply:
Maybe let's do it like this: We have to queries, one with watch date sorting (where we have to fetch all movies every time) and one without. We only use the watch date sorting query when necessary. This way we can lessen the impact of this performance issue.

Reply:
to be fair this change would complicate the way sort-by is handled now, and the performance issue would still be there for everyone who sorts by watch date.

i feel like it is a generally negligible performance issue for the scope of this query, because you would need a LOT of entries for it to slow down the response time. we could of course run some tests with this but i thinks we should stick with the current way that sort-by is handled to avoid the added complexity just for one more way to sort movies.

Reply:
just to make sure, could you try using the latest commit and see if you still get ghost entries

Reply:
The ghost entries seem gone to me :heavy_check_mark: 

I just checked the query, on my newish pc the original query takes ~0.016 and the new one 0.025 (user had 860 watch dates). I am not sure how well this scales with increased traffic and/or bigger libraries, especially on more low level hardware.
True, probably nothing to worry about most of the time, but still sucks :D

Reply:
nice to see its working.

yeah the performance decrease is pretty much negligible, especially considering that 860 watch dates is a lot for most people. for now it can be something to keep in mind if problems about performance arise in the future

Reply:
I saw people with 1-2k entries when debugging trakt issues, some even more, you should not underestimate how people use this. but this are of course still low numbers.